src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,42 +1,84 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { User } from '../../types/User';
+import { getUser } from '../../api';
+import { useAppSelector } from '../../app/hooks';
+import classNames from 'classnames';
+import { useDispatch } from 'react-redux';
+import { setCurrentTodo } from '../../features/currentTodo';
 
 export const TodoModal: React.FC = () => {
+  const [isModalLoading, setIsModalLoading] = useState(false);
+  const [user, setUser] = useState<User>();
+
+  const currentTodo = useAppSelector(state => state.currentTodo);
+
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    setIsModalLoading(true);
+
+    if (!currentTodo) {
+      return;
+    }
+
+    getUser(currentTodo.userId)
+      .then(data => setUser({ ...data }))
```

Comment:
It's a good practice to handle errors for API calls. Consider adding a .catch block to handle any potential errors that might occur during the API call.